src/libstd/sys/unix/backtrace/tracing/gcc_s.rs

```diff
@@ -35,28 +36,40 @@ pub fn write(w: &mut Write) -> io::Result<()> {
     unsafe {
         LOCK.lock();
 
-        writeln!(w, "stack backtrace:")?;
-
-        let mut cx = Context { writer: w, last_error: None, idx: 0 };
-        let ret = match {
-            uw::_Unwind_Backtrace(trace_fn,
-                                  &mut cx as *mut Context as *mut libc::c_void)
-        } {
-            uw::_URC_NO_REASON => {
+        let mut cx = Context {
+            last_error: None,
+            idx: 0,
+            frames: frames,
+        };
+        let result_unwind = uw::_Unwind_Backtrace(trace_fn,
+                                                  &mut cx as *mut Context
+                                                          as *mut libc::c_void);
+        // See libunwind:src/unwind/Backtrace.c for the return values.
+        // No, there is no doc.
+        let ret = match result_unwind {
+            uw::_URC_END_OF_STACK => {
```

Comment:
Hm I feel like we're still not getting to an answer to my question. The code you linked to ignores errors because it has nowhere to pass the errors to and the information is just advisory, so it's fine to just ignore errors.

My question is *why* did this change happen? Is it purely being proactive? Was it reactive to some problem?